.github/workflows/docker-publish.yml

```diff
@@ -29,10 +29,10 @@ jobs:
       - uses: Swatinem/rust-cache@v2
         with:
           cache-on-failure: true
-      - name: Install cross
-        id: cross
+      - name: Install cross main
+        id: cross_main
         run: |
-          cargo install cross
+          cargo install cross --git https://github.com/cross-rs/cross
       # Login against a Docker registry except on PR
```

Comment:
Is it strictly necessary to install this directly from Git for the Docker image?

I'm OK with merging this as it unblocks users now but imo we should not install directly from Git without commit pinning (`--rev <commit-hash>`)

Context: https://www.stepsecurity.io/blog/harden-runner-detection-tj-actions-changed-files-action-is-compromised

Reply:
that's a good point, this was done to fix https://github.com/foundry-rs/foundry/pull/10035 however I can see ring fixed their issue so should not install from main anymore, going to revert to 0.2.5 and see if it all OK

Reply:
@zerosnacks neah, this still fails https://github.com/foundry-rs/foundry/actions/runs/13922513989/job/38958875948 because fix was included in a newer version of ring and cargo update is reverted in this commit. should we just pin a rev for now?

Reply:
it is now reverted